src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -59,31 +59,83 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+    // - Detects if any of the user's actions are identical to the inbox actions,
+    //   and if so, deletes them and redirects their keybindings to the inbox actions
+    // - We have to do this here instead of when loading since we don't actually have
+    //   any parents while loading the user settings, the parents are added after
+    void ActionMap::_FinalizeInheritance()
+    {
+        // first, gather the inbox actions from the relevant parent
+        std::unordered_map<InternalActionID, Model::Command> inboxActions;
+        winrt::com_ptr<implementation::ActionMap> foundParent{ nullptr };
+        for (const auto& parent : _parents)
         {
-            // ActionMap should never point to nullptr
-            FAIL_FAST_IF_NULL(maskingPair->second);
+            for (const auto& [_, cmd] : parent->_ActionMap)
+            {
+                if (cmd.Origin() != OriginTag::InBox)
+                {
+                    // only one parent contains all the inbox actions and that parent contains only inbox actions,
+                    // so if we found a non-inbox action we can just skip to the next parent
+                    continue;
+                }
+                foundParent = parent;
+                break;
+            }
+        }
 
-            // masking actions cannot contain nested or invalid commands,
-            // so we can just return it directly.
-            return maskingPair->second;
+        if (foundParent)
+        {
+            for (const auto& [_, cmd] : foundParent->_ActionMap)
+            {
+                inboxActions.emplace(Hash(cmd.ActionAndArgs()), cmd);
+            }
         }
 
+        // now, look through our _ActionMap for commands that
+        // - had an ID generated for them
+        // - do not have a name/icon path
+        // - have a hash that matches a command in the inbox actions
+        std::erase_if(_ActionMap, [&](const auto& pair) {
+            const auto userCmdImpl{ get_self<Command>(pair.second) };
+            if (userCmdImpl->IdWasGenerated() && !userCmdImpl->HasName() && userCmdImpl->IconPath().empty())
+            {
+                const auto userActionHash = Hash(userCmdImpl->ActionAndArgs());
+                if (const auto inboxCmd = inboxActions.find(userActionHash); inboxCmd != inboxActions.end())
+                {
+                    for (const auto& [key, cmdID] : _KeyMap)
+                    {
+                        // for any of our keys that point to the user action, point them to the inbox action instead
+                        if (cmdID == pair.first)
+                        {
+                            _KeyMap.insert_or_assign(key, inboxCmd->second.ID());
+
+                            // register the keys with the inbox action
+                            inboxCmd->second.RegisterKey(key);
+                        }
+                    }
+
+                    // remove this pair
+                    return true;

```

Comment:
this sounds like a fixup -- but it looks like we don't know that or set `fixupsAppliedDuringLoad`

Reply:
Since `IDWasGenerated` is true, `fixupsAppliedDuringLoad` has already been set